src/app/portlets/dot-edit-content/components/dot-edit-page-toolbar/dot-edit-page-toolbar.component.spec.ts

```diff
@@ -87,7 +130,95 @@ describe('DotEditPageToolbarComponent', () => {
 
         fixture.detectChanges();
 
-        const primaryAction: DebugElement = de.query(By.css('.edit-page-toolbar__primary-action'));
+        const primaryAction: DebugElement = de.query(By.css('.edit-page-toolbar__save'));
         expect(primaryAction.nativeElement.disabled).toBeFalsy('the save button have to be enable');
     });
+
+    it('should turn on the locker and emit page state and lock state when user click on edit mode', () => {
+        fixture.detectChanges();
+        const editButton = de.queryAll(By.css('.edit-page-toolbar__state-selector .ui-button'))[0].nativeElement;
+
+        let pageStateResult;
+        component.pageState.subscribe((res) => {
+            pageStateResult = res;
+        });
+
+        let lockPageResult;
+        component.lockPage.subscribe((res) => {
+            lockPageResult = res;
+        });
+
+        editButton.click();
+
+        expect(component.pageLocked).toBe(true, 'lock page');
+        expect(lockPageResult).toBe(true, 'lock page output emitted');
+        expect(pageStateResult).toEqual('edit', 'page state output emitted');
+    });
+
+    it('should keep the locker true while moving through states', () => {
+        component.pageLocked = true;
+        fixture.detectChanges();
+
+        expect(component.stateSelected).toEqual('preview', 'preview state by default');
+
+        const stateSelectorButtons: DebugElement[] = de.queryAll(
+            By.css('.edit-page-toolbar__state-selector .ui-button')
+        );
+        const editButton = stateSelectorButtons[0].nativeElement;
+        const previewButton = stateSelectorButtons[1].nativeElement;
+        const liveButton = stateSelectorButtons[2].nativeElement;
+
+        let pageStateResult;
+        component.pageState.subscribe((res) => {
+            pageStateResult = res;
+        });
+
+        let lockPageResult;
+        component.lockPage.subscribe((res) => {
+            lockPageResult = res;
+        });
+
+        editButton.click();
```

Comment:
maybe we could create a util test function to this something like:

function clickEditButton() {
     const editButton = de.queryAll(By.css('.edit-page-toolbar__state-selector .ui-button'))[0].nativeElement;
     const editButton = stateSelectorButtons[0].nativeElement;
    editButton.click();
}

if later we change the selector 'edit-page-toolbar__state-selector', we should just make change in one place into the test